ir/src/lib.rs

```diff
@@ -168,6 +182,58 @@ impl AirIR {
     }
 }
 
+// === HELPERS ====================================================================================
+
+/// Returns an error if a public inputs section is not defined.
+fn validate_public_inputs(public_inputs: &PublicInputs) -> Result<(), SemanticError> {
+    if public_inputs.is_empty() {
+        return Err(SemanticError::MissingSection(
+            "Public Inputs section is missing".to_string(),
```

Comment:
nit: Similar comment to the one above for the `trace_columns`, and same thing for boundary constraints and transition constraints below

Reply:
Fixed, thanks.